contribs/src/main/java/com/netflix/conductor/contribs/http/RestClientManager.java

```diff
@@ -30,9 +31,27 @@
 @Singleton
 public class RestClientManager {
 
-	private Client defaultClient = Client.create();
-	
-	public Client getClient(Input input) {
-		return defaultClient;
+	static final int DEFAULT_READ_TIMEOUT = 150;
+	static final int DEFAULT_CONNECT_TIMEOUT = 100;
+	static final String HTTP_TASK_READ_TIMEOUT = "http.task.read.timeout";
+	static final String HTTP_TASK_CONNECT_TIMEOUT = "http.task.connect.timeout";
+
+
+	private final ThreadLocal<Client> threadLocalClient;
+	private final int defaultReadTimeout;
+	private final int defaultConnectTimeout;
+
+	public RestClientManager(Configuration config) {
+		this.threadLocalClient = ThreadLocal.withInitial(()->Client.create());
```

Comment:
Given this class is Singleton, not sure how ThreadLocal is helpful here.

Reply:
We are mutating the Client object in the HttpTask.java, httpCall method. As this method can be called from multiple threads, I have tried to make the Client object threadlocal. Please let me know if I am missing something.

Reply:
@kishorebanala  Can you please let me know your thoughts.